.github/workflows/pre-merge-integration-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Pre-merge integration test
+on:
+  workflow_run:
+    workflows: ["Pre-merge checks"]
+    types:
+      - completed
+
+jobs:
+  deploy:
+    name: pre-merge-integration-tests
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-dev
+    timeout-minutes: 15
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: dev
+      STACK_NAME: address-cri-api
+      ROLLBACK_ACTION: "--disable-rollback"
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.GH_ACTIONS_ROLE_ARN }}
```

Comment:
We should not share accounts for code that is part of the production pipeline and code that has yet to be reviewed because of the risk of a vulnerability being created that is then used to corrupt the production pipeline. i.e. build and "dev" should be different accounts.

Reply:
I think they are different accounts, as their [github environment ](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)is different -  @mrdoodles can you confirm?